smartapps/ethayer/lock-user.src/lock-user.groovy

```diff
@@ -571,6 +573,14 @@ def isActiveKeypad() {
   }
 }
 
+def isUserEnabled() {
```

Comment:
I made the changes you suggested. Not sure if I'm missing anything else. I've been using the previous version for my own setup with no issues, but I wouldn't call it throughly tested.